hazelcast/src/main/java/com/hazelcast/query/impl/Indexes.java

```diff
@@ -173,6 +173,16 @@ public synchronized InternalIndex addOrGetIndex(IndexConfig indexConfig) {
         converterCache.invalidate(index);
 
         indexes = indexesByName.values().toArray(EMPTY_INDEXES);
+        /**
+         * Sort indexes by creation timestamp. Some high-level
+         * sub-systems like HD SQL optimizer may need the oldest index that has the most
+         * chances to be completely constructed and being usable for queries.
+         */
+        Arrays.sort(indexes, (o1, o2) -> {
```

Comment:
The array should be sorted before exposing it through the volatile `indexes` field to avoid inconsistent view into the array from other threads. Basically, once `indexes` field is written the array should be treated as immutable.

Reply:
Good catch, fixed